src/containers/Layout/Layout.tsx

```diff
@@ -19,12 +19,17 @@ import MenuButton from '#components/MenuButton/MenuButton';
 import UserMenu from '#components/UserMenu/UserMenu';
 import { addQueryParam } from '#src/utils/location';
 import { getSupportedLanguages } from '#src/i18n/config';
+import { useProfileStore } from '#src/stores/ProfileStore';
+import { unpersistProfile, useListProfiles, useProfilesFeatureEnabled } from '#src/hooks/useProfiles';
 
 const Layout = () => {
   const location = useLocation();
   const navigate = useNavigate();
   const { t, i18n } = useTranslation('common');
   const { config, accessModel } = useConfigStore(({ config, accessModel }) => ({ config, accessModel }), shallow);
+  const { data: profilesData } = useListProfiles();
+  const profiles = profilesData?.responseData.collection;
+  const profilesEnabled = useProfilesFeatureEnabled();
```

Comment:
Why do we need to fetch profiles and  check profilesEnabled data separately in various parts of our codebase? What do you think of centralizing the logic to handle profilesData queries based on the profilesEnabled?